measurements/api/measurements.py

```diff
@@ -420,171 +390,226 @@ def list_measurements(
 
     if since is not None:
-        fpq = fpq.filter(Fastpath.measurement_start_time > since)
+        query_params["since"] = since
+        mrwhere.append(sql.text("measurement.measurement_start_time > :since"))
+        fpwhere.append(sql.text("measurement_start_time > :since"))
+
     if until is not None:
-        fpq = fpq.filter(Fastpath.measurement_start_time <= until)
+        query_params["until"] = until
+        mrwhere.append(sql.text("measurement.measurement_start_time <= :until"))
+        fpwhere.append(sql.text("measurement_start_time <= :until"))
+
     if report_id:
-        fpq = fpq.filter(Fastpath.report_id == report_id)
+        query_params["report_id"] = report_id
+        mrwhere.append(sql.text("report.report_id = :report_id"))
+        fpwhere.append(sql.text("report_id = :report_id"))
+
     if probe_cc:
-        fpq = fpq.filter(Fastpath.probe_cc == probe_cc)
+        query_params["probe_cc"] = probe_cc
+        mrwhere.append(sql.text("report.probe_cc = :probe_cc"))
+        fpwhere.append(sql.text("probe_cc = :probe_cc"))
+
     if probe_asn is not None:
-        fpq = fpq.filter(Fastpath.probe_asn == probe_asn)
+        query_params["probe_asn"] = probe_asn
+        mrwhere.append(sql.text("report.probe_asn = :probe_asn"))
+        fpwhere.append(sql.text("probe_asn = :probe_asn"))
+
     if test_name is not None:
-        fpq = fpq.filter(Fastpath.test_name == test_name)
-    if input_:
-        fpq = fpq.filter(Fastpath.input == input_)
-    elif domain:
-        if test_name in [None, "web_connectivity", "http_requests"]:
-            fpq = fpq.filter(
-                or_(
-                    Fastpath.input.startswith(domain),
-                    Fastpath.input.startswith("http://" + domain),
-                    Fastpath.input.startswith("https://" + domain),
-                )
-            )
-        else:
-            fpq = fpq.filter(Fastpath.input.startswith(domain))
+        query_params["test_name"] = test_name
+        mrwhere.append(sql.text("report.test_name = :test_name"))
+        fpwhere.append(sql.text("test_name = :test_name"))
 
     # TODO, we don't support filtering by confirmed filter in the fastpath, so
     # we exclude all the fastpath measurements when filtering by confirmed is
     # set
     if confirmed is not None:
-        fpq = fpq.filter(False)
+        query_params["confirmed"] = confirmed
+        # TODO: coalesce false() ?
+        mrwhere.append(sql.text("measurement.confirmed = :confirmed"))
+        fpwhere.append(sql.text("false"))
+
     if anomaly is not None:
-        fpq = fpq.filter(
-            cast(cast(Fastpath.scores["blocking_general"], String), Float) > 0.5
+        # TODO: coalesce false() ?
+        query_params["anomaly"] = anomaly
+        mrwhere.append(sql.text("measurement.anomaly = :anomaly"))
+        # fpwhere.append(sql.text("scores->>'blocking_general' > 0.5"))
+        fpwhere.append(sql.text("false"))
```

Comment:
I think we should add a comment here saying that currently we exclude from the filter the fastpath table as it doesn't have a confirmed or anomaly column and perhaps reference the relevant issue.

Reply:
It's going to be fixed very soon in a new PR. I'll reference this one.

Reply:
I filed this issue: https://github.com/ooni/api/issues/157